library/src/main/java/com/xfinity/blueprint/view/ScreenViewDelegate.kt

```diff
@@ -112,9 +114,28 @@ open class ScreenViewDelegate(componentRegistry: ComponentRegistry,
 
         componentAdapter.clear(false)
         componentAdapter.addComponents(newComponents, false)
-        diff.dispatchUpdatesTo(componentAdapter)
+
+        val updateBehavior = {
+            diff.dispatchUpdatesTo(componentAdapter)
+        }
+
+        if (!recyclerView.isComputingLayout) {
+            updateBehavior.invoke()
+        } else {
+            //If we dipatch updates while the recyclerview is computing, the app will crash
+            recyclerView.addOnScrollListener(OnIdleScrollListener(updateBehavior))
+        }
     }
 
+    private inner class OnIdleScrollListener(private val behavior: () -> Unit) : RecyclerView.OnScrollListener() {
```

Comment:
This could be an inline class for a performance boost but it's still experimental